src/test/java/com/getindata/connectors/http/internal/table/lookup/querycreators/GenericJsonQueryCreatorTest.java

```diff
@@ -52,6 +52,23 @@ public void shouldSerializeToJson() {
         row.setField(0, 11);
         row.setField(1, StringData.fromString("myUuid"));
 
+        this.jsonQueryCreator.createLookupQuery(row);
+        LookupQueryInfo lookupQuery = this.jsonQueryCreator.createLookupQuery(row);
+        assertThat(lookupQuery.getBodyBasedUrlQueryParameters().isEmpty());
+        assertThat(lookupQuery.getLookupQuery()).isEqualTo("{\"id\":11,\"uuid\":\"myUuid\"}");
+    }
+
+    @Test
+    public void shouldSerializeToJsonTwice() {
+        GenericRowData row = new GenericRowData(2);
+        row.setField(0, 11);
+        row.setField(1, StringData.fromString("myUuid"));
+
+        this.jsonQueryCreator.createLookupQuery(row);
+
+        // Call createLookupQuery two times
+        // to check that serialization schema is not opened Two times.
+        this.jsonQueryCreator.createLookupQuery(row);
```

Comment:
Tests suggests method is called twice. I can see however three calls to `this.jsonQueryCreator.createLookupQuery(row);` What am I missing?

Reply:
Right. I removed one `createLookupQuery` call.